edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -10,13 +10,18 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Docker
     using Newtonsoft.Json.Linq;
     using static System.FormattableString;
 
+
     [JsonConverter(typeof(DockerConfigJsonConverter))]
     public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
+        // Match for an environment variable in the official POSIX format $ + "letter" + "letter|digit"
+        // Then the port has to be specified with ":"
+        const string ImageUpstreamRegexPattern = @"^\$upstream(?<path>:[1-9].*)";
```

Comment:
check if it's prefix is "$upstream" and replace with environment variable. Should support $upstream for any case.

Reply:
should be easier to use simple string.startsWith.

Reply:
I did it as explained above, I ll rather keep using image[0]

Reply:
would you tell me the benefit using regex over startsWith method call in this case?

Reply:
Does 2 things:
1. checks for a port
2. Can generate exception if user made typos in $upstream